packages/cli/src/commands/source/push/add.ts

```diff
@@ -0,0 +1,154 @@
+
+  public static flags = {
+    file: flags.string({
+      multiple: true,
+      exclusive: ['folder'],
+      char: 'f',
+      helpValue: 'myfile.json',
+      description: 'One or multiple file to push. Can be repeated.',
+    }),
+    folder: flags.string({
+      multiple: true,
+      exclusive: ['file'],
+      char: 'd',
+      helpValue: './my_folder_with_multiple_json_files',
+      description:
+        'One or multiple folder containing json files. Can be repeated',
+    }),
+  };
+
+  public static args = [
+    {
+      name: 'sourceId',
+      required: true,
+      description:
+        'The identifier of the source on which to perform the add operation. See source:push:list to obtain the identifier.',
+    },
+  ];
+
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    const {flags} = this.parse(SourcePushAdd);
+    if (!flags.file && !flags.folder) {
+      this.error(
+        'You must provide either --file= or --folder=. See source:push:add --help for more information.'
+      );
+    }
+    const cfg = await new AuthenticatedClient().cfg.get();
+    const source = new Source(cfg.accessToken!, cfg.organization);
+
+    if (flags.file) {
+      await this.doPushFile(source);
+    }
+    if (flags.folder) {
+      await this.doPushFolder(source);
+    }
+
+    await this.config.runHook(
+      'analytics',
+      buildAnalyticsSuccessHook(this, flags)
+    );
+  }
+
+  private async doPushFolder(source: Source) {
+    const {flags, args} = this.parse(SourcePushAdd);
+
+    const fileNames = flags.folder.flatMap((folder) => {
+      const files = readdirSync(folder);
+      return files.map((f) => `${path.join(folder, f)}`);
+    });
+
+    const docBuilders = flags.folder.flatMap((folder) => {
+      return readdirSync(folder).flatMap((file) => {
+        const fullPath = path.join(folder, file);
+        const docBuilders =
+          parseAndGetDocumentBuilderFromJSONDocument(fullPath);
+        this.successMessageOnParseFile(fullPath, docBuilders.length);
+        return docBuilders;
+      });
+    });
+
+    const res = await source.batchUpdateDocuments(args.sourceId, {
+      addOrUpdate: docBuilders,
+      delete: [],
+    });
+    this.successMessageOnAdd(fileNames, docBuilders.length, res);
+  }
+
+  private async doPushFile(source: Source) {
+    const {flags, args} = this.parse(SourcePushAdd);
+
+    const docBuilders = flags.file.flatMap((file) => {
+      const docBuilders = parseAndGetDocumentBuilderFromJSONDocument(file);
+      this.successMessageOnParseFile(file, docBuilders.length);
+      return docBuilders;
+    });
+
+    const res = await source.batchUpdateDocuments(args.sourceId, {
+      addOrUpdate: docBuilders,
+      delete: [],
+    });
+    this.successMessageOnAdd(flags.file, docBuilders.length, res);
+  }
+
+  private successMessageOnAdd(
+    files: string[],
+    numAdded: number,
+    res: AxiosResponse
+  ) {
+    let fileNames = files.slice(0, 5).join(', ');
```

Comment:
That's a bit cryptic, what the goal of this line?

Reply:
I can add a comment for explanation.